test/e2e/main_test.go

```diff
@@ -90,6 +96,206 @@ type InstallTestCase struct {
+						},
+						Image: &metering.ImageConfig{},
+						Config: &metering.ReportingOperatorConfig{
+							LogLevel: "debug",
+							Prometheus: &metering.ReportingOperatorPrometheusConfig{
+								MetricsImporter: &metering.ReportingOperatorPrometheusMetricsImporterConfig{
+									Config: &metering.ReportingOperatorPrometheusMetricsImporterConfigSpec{
+										ChunkSize:                 &meta.Duration{Duration: 5 * time.Minute},
+										PollInterval:              &meta.Duration{Duration: 30 * time.Second},
+										StepSize:                  &meta.Duration{Duration: 1 * time.Minute},
+										MaxImportBackfillDuration: &meta.Duration{Duration: 15 * time.Minute},
+										MaxQueryRangeDuration:     &meta.Duration{Duration: 5 * time.Minute},
+									},
+								},
+							},
+						},
+					},
+				},
+				Presto: &metering.Presto{
+					Spec: &metering.PrestoSpec{
+						Coordinator: &metering.PrestoCoordinatorSpec{
+							Resources: &v1.ResourceRequirements{
+								Requests: v1.ResourceList{
+									v1.ResourceCPU:    resource.MustParse("1"),
+									v1.ResourceMemory: resource.MustParse("1Gi"),
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	for _, testCase := range tt {
+		t := t
+		testCase := testCase
+
+		if testCase.Skip {
+			continue
+		}
+
+		t.Run(testCase.Name, func(t *testing.T) {
+			testManualOLMUpgradeInstall(
+				t,
+				testCase.Name,
+				namespacePrefix,
+				testCase.MeteringOperatorImageRepo,
+				testCase.MeteringOperatorImageTag,
+				testOutputPath,
+				testCase.ExpectInstallErrMsg,
+				testCase.ExpectInstallErr,
+				testCase.PurgeReports,
+				testCase.PurgeReportDataSources,
+				testCase.InstallSubTest,
+				testCase.MeteringConfigSpec,
+			)
+		})
+	}
+}
+
+func testManualOLMUpgradeInstall(
+	t *testing.T,
+	testCaseName,
+	namespacePrefix,
+	meteringOperatorImageRepo,
+	meteringOperatorImageTag,
+	testOutputPath string,
+	expectInstallErrMsg []string,
+	expectInstallErr,
+	purgeReports,
+	purgeReportDataSources bool,
+	testInstallFunction InstallTestCase,
+	testMeteringConfigSpec metering.MeteringConfigSpec,
+) {
+	// create a directory used to store the @testCaseName container and resource logs
+	testCaseOutputBaseDir := filepath.Join(testOutputPath, testCaseName)
+	err := os.Mkdir(testCaseOutputBaseDir, 0777)
+	require.NoError(t, err, "creating the test case output directory should produce no error")
+
+	// create a pre-upgrade test case directory
+	preUpgradeTestOutputDir := filepath.Join(testCaseOutputBaseDir, preUpgradeTestDirName)
+	err = os.Mkdir(preUpgradeTestOutputDir, 0777)
+	require.NoError(t, err, "creating the test case output directory should produce no error")
+
+	testFuncNamespace := fmt.Sprintf("%s-%s", namespacePrefix, strings.ToLower(testCaseName))
+	if len(testFuncNamespace) > kubeNamespaceCharLimit {
+		require.Fail(t, "The length of the test function namespace exceeded the kube namespace limit of %d characters", kubeNamespaceCharLimit)
```

Comment:
How about just warn on this? Kube will just truncate and carry on, won't it?

Reply:
Unfortunately not, as attempting to create the namespace in the `deploy` package will error if the total number of characters in the namespace, exceeds the namespace character limit.

Reply:
ah